rustls/src/crypto/ring/sign.rs

```diff
@@ -212,16 +212,12 @@ impl EcdsaSigningKey {
             _ => unreachable!(), // all callers are in this file
         };
 
-        // wrap sec1 encoding in an OCTET STRING
-        let mut sec1_wrap = Vec::with_capacity(maybe_sec1_der.len() + 8);
-        sec1_wrap.extend_from_slice(maybe_sec1_der);
-        wrap_in_asn1_len(&mut sec1_wrap);
-        sec1_wrap.insert(0, der::Tag::OctetString as u8);
+        let sec1_wrap = asn1_wrap(der::Tag::OctetString as u8, maybe_sec1_der);
 
-        let mut pkcs8 = Vec::with_capacity(pkcs8_prefix.len() + sec1_wrap.len() + 4);
-        pkcs8.extend_from_slice(pkcs8_prefix);
-        pkcs8.extend_from_slice(&sec1_wrap);
-        wrap_in_sequence(&mut pkcs8);
+        let mut pkcs8_inner = Vec::with_capacity(pkcs8_prefix.len() + sec1_wrap.len());
+        pkcs8_inner.extend_from_slice(pkcs8_prefix);
+        pkcs8_inner.extend_from_slice(&sec1_wrap);
+        let pkcs8 = wrap_in_sequence(&pkcs8_inner);
```

Comment:
nit: it seems like this `pkcs8` binding doesn't add much. Maybe in-line on L222 in the `from_pkcs8` call?